tgbot/src/request/form.rs

```diff
@@ -0,0 +1,72 @@
+use crate::types::{InputFile, InputFileInfo, InputFileKind, InputFileReader};
+use hyper_multipart_rfc7578::client::multipart::Form as MultipartForm;
+use std::collections::HashMap;
+
+#[derive(Debug)]
+pub(crate) enum FormValue {
+    Text(String),
+    File(InputFile),
+}
+
+impl<T> From<T> for FormValue
+where
+    T: ToString,
+{
+    fn from(value: T) -> Self {
+        FormValue::Text(value.to_string())
+    }
+}
+
+impl From<InputFile> for FormValue {
+    fn from(value: InputFile) -> Self {
+        FormValue::File(value)
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct Form {
+    fields: HashMap<String, FormValue>,
+}
+
+impl Form {
+    pub(crate) fn new() -> Self {
+        Self { fields: HashMap::new() }
+    }
+
+    pub(crate) fn set_field<N, V>(&mut self, name: N, value: V)
+    where
+        N: Into<String>,
+        V: Into<FormValue>,
+    {
+        self.fields.insert(name.into(), value.into());
+    }
+
+    pub(crate) fn into_multipart(self) -> MultipartForm<'static> {
```

Comment:
Why not implement `From<Form> for MultipartForm<'static>`?

Reply:
done